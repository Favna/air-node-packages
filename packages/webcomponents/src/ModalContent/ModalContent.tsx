import css from './ModalContent.scss';
import React, { FC, memo, PropsWithChildren } from 'react';

/** Creates modal content using pre-defined Rijkswatestaat styling */
export const ModalContent: FC<PropsWithChildren<unknown>> = props => (
  <div className={css.content}>
    {props.children}
  </div>
);

export default memo(ModalContent);