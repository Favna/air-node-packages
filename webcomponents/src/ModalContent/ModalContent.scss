@import '../styles/vars';

.content > * {
  @include font-normal;
  color: $text-black;
  font-family: $rws-font-text !important;
  font-size: 1rem;
  letter-spacing: 0.6px;
  line-height: 1.81;
}

.content {
  padding-top: 0.2rem;
}
