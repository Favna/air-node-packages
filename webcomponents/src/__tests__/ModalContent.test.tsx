import { shallow, ShallowWrapper } from 'enzyme';
import ModalContent from '../ModalContent/ModalContent';
import React from 'react';

let modalContent: ShallowWrapper;

describe('Snapshot Testing', () => {
  test('Required Props', () => {
    const modalContent = shallow(<ModalContent />);
    expect(modalContent).toMatchSnapshot();
  });

  test('Optional Props', () => {
    const text = 'The Answer is 42';
    beforeAll(() => {
      modalContent = shallow(
        <ModalContent>
          <p>{text}</p>
        </ModalContent>
      );
    })
    expect(modalContent).toMatchSnapshot();
  });
});

describe('With Children', () => {
  beforeAll(() => {
    modalContent = shallow(<ModalContent />);
  });

  test('should match snapshot', () => {
    expect(modalContent).toMatchSnapshot();
  });

  test('should render without crashing', () => {
    expect(modalContent).toBeTruthy();
  });

  test('should render empty div when no children are given', () => {
    expect(modalContent.prop('children')).toBeUndefined();
  });

});

describe('With Children', () => {
  const text = 'The Answer is 42';
  beforeAll(() => {
    modalContent = shallow(
      <ModalContent>
        <p>{text}</p>
      </ModalContent>
    );
  });

  test('should match snapshot', () => {
    expect(modalContent).toMatchSnapshot();
  });

  test('should insert children when getting any', () => {
    expect(modalContent.prop('children')).toBeDefined();
    expect(modalContent.find('p').text()).toBe(text);
  });
});