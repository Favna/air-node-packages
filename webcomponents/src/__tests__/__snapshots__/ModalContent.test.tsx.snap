// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Children should match snapshot 1`] = `<div />`;

exports[`With Children should match snapshot 2`] = `
<div>
  <p>
    The Answer is 42
  </p>
</div>
`;
